### ISSUE TITLE
Sometime Inventory Valuation Report give error dialog The cost of [Product] is currently equal to 0

### ISSUE THREAD

**Issue by Arvinman** [odoo/odoo#24973]
Impacted versions:
Odoo 11:

Steps to reproduce:
- Inventory/Reporting/Inventory Valuation/Current Inventory

Current behavior:
- Sometime the report will give error dialog "The cost of [Product] is currently equal to 0. Change the cost or the configuration of your product to avoid an incorrect valuation." even the cost of it isn't 0.
- Probably a rounding issue error. 

    In stock_account/models/stock.py  [def _fifo_vacuum(self)]

            corrected_value = remaining_value_before_vacuum + tmp_value
            move.write({
                'remaining_value': new_remaining_value,
                'remaining_qty': new_remaining_qty,
                'value': move.value - corrected_value,
            })

            if move.product_id.valuation == 'real_time':
                # If `corrected_value` is 0, absolutely do *not* call `_account_entry_move`. We
                # force the amount in the context, but in the case it is 0 it'll create an entry
                # for the entire cost of the move. This case happens when the candidates moves
                # entirely compensate the problematic move.
                if not corrected_value:
                    continue

In my case 
- **remaining_value_before_vacuum** = -10667.72
- **tmp_value** = 10667.720000000001
thus **corrected_value** became 1.8189894035458565e-12 but not zero. So it is not handled by **if not corrected_value:** 
 
Expected behavior:
- Inventory Valuation should give us report without error

**Comment by bazemoreb** [comment]
Same issue here but getting the error on trying to receive a product in to inventory.

**Comment by pedrobaeza** [comment]
@jco-odoo I think instead `if not corrected_value:`, it should be `if not float_is_zero(corrected_value, ..):`

**Comment by jco-odoo** [comment]
Sure this has not been corrected already?  @sle-odoo

**Comment by pedrobaeza** [comment]
I still see the same code:

https://github.com/odoo/odoo/blob/052a5702c500a745fa94d13244eeaf14baccecb8/addons/stock_account/models/stock.py#L451

**Comment by bazemoreb** [comment]
Just clarifying on my report of same issue here.  We were able to figure out what happened, the error message was a bit misleading is all.  Our product had a cost, but the PURCHASE ORDER had been zero'd out on that product manually.  So the actual problem was that on the PO, we had a product with zero dollars, and odoo didn't like this.  This must have been a change in how things work (and totally makes sense).  We've adjusted our internal process to handle, but in case anyone stumbles on this, verify that the product itself has a >0 cost, and also the associated purchase order.
